### PR TITLE
 Retrieve status for pods in pods controller.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
Retrieve the status for one pod. It seems that the current implementation as resource
is wrong.

JIRA issues: MARATHON-7878

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala [hunk 1]
```diff
@@ -147,7 +149,23 @@ class PodsController(
...
+  def status(podId: PathId): Route =
+    authenticated.apply { implicit identity =>
+      podManager.find(podId) match {
+        case None =>
+          reject(Rejections.EntityNotFound.noPod(podId))
+        case Some(pod) =>
+          authorized(ViewRunSpec, pod).apply {
+            onSuccess(podStatusService.selectPodStatus(podId)) { maybeStatus =>
+              // If selectPodStatus returns None this is a bug since find(podId) already verifies that the pod exists.
+              // We don't filter the pods with an authentication since we check for authorization before.
```
Comment (alenkacz) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: authentication or authorization? I think you wanted to use word authorization in both cases in this sentence, right?

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala [hunk 2]
```diff
@@ -147,7 +149,23 @@ class PodsController(
...
+  @SuppressWarnings(Array("all")) // async/await
+  def status(podId: PathId): Route =
+    authenticated.apply { implicit identity =>
+      podManager.find(podId) match {
+        case None =>
+          reject(Rejections.EntityNotFound.noPod(podId))
+        case Some(pod) =>
+          authorized(ViewRunSpec, pod).apply {
+            onSuccess(podStatusService.selectPodStatus(podId)) { maybeStatus =>
+              // If selectPodStatus returns None this is a bug since find(podId) already verifies that the pod exists.
```
Comment (alenkacz) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: is it possible that the pod might be deleted by another request in between find and select calls?

Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: I think so. However, IMHO that is still a bug in our setup. It's actually an issue with `selectPodStatus` which does not distinguish between pod not found or access rejected.

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala (hunk 1 above)
Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: It should be "authorization". Thanks.

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala [hunk 3]
```diff
@@ -147,7 +149,23 @@ class PodsController(
...
+              // We don't filter the pods with an authentication since we check for authorization before.
+              val status: raml.PodStatus = maybeStatus.get
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala]: Good point.

**Pull request merged**